### PR TITLE
feat: simplify and optimise

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
## Dockerfile 

* removes unused stuff
* Correctly sets `$HOME`
* Plugin cache dir set

## Scripts

* New `clean` script to ensure sane environment prior to running tests and to tidy up afterwards

## Makefile

* clean target (and clean2) so we can ensure sane environment and clean up afterwards

### CODE REVIEW

File: avmmakefile [hunk 1]
```diff
@@ -80,4 +80,10 @@ autofix:
 grept-apply:
 	curl -H 'Cache-Control: no-cache, no-store' -sSL "$(REMOTE_SCRIPT)/grept-apply.sh" | bash
 
-.PHONY: docs docscheck fmt gofmt fumpt gosec tffmtcheck tfvalidatecheck terrafmtcheck gofmtcheck golint tflint lint checkovcheck checkovplancheck fmtcheck pr-check unit-test e2e-test version-upgrade-test terrafmt pre-commit depsensure yor-tag autofix tools
+clean:
+	curl -H 'Cache-Control: no-cache, no-store' -sSL "$(REMOTE_SCRIPT)/clean.sh" | bash
+
+clean2:
```
Comment (jaredfholgate) [inline avmmakefile:86]: Aren't clean and clean2 the same?

File: avmmakefile [hunk 2]
```diff
@@ -49,7 +49,7 @@ checkovplancheck:
 
 fmtcheck: gofmtcheck tffmtcheck terrafmtcheck
 
-pr-check: fmtcheck tfvalidatecheck lint unit-test
+pr-check: clean fmtcheck tfvalidatecheck lint unit-test clean2
```
Comment (jaredfholgate) [inline avmmakefile:52]: Won't this impact people developing and testing locally? Deleting their state file may not be expected. I guess people should not be running this locally though.

File: Dockerfile [hunk 3]
```diff
@@ -97,13 +95,17 @@ RUN yum update -y && \
...
+COPY --from=build /src/tfenv ${HOME_DIR}/tfenv
+COPY --from=build /src/pkenv ${HOME_DIR}/pkenv
+RUN mkdir ${HOME_DIR}/tflintenv && \
+    mkdir -p ${HOME_DIR}/.terraform.d/plugin-cache && \
+    chmod -Rv a+rwX ${HOME_DIR} && \
+    chmod 777 ${HOME_DIR}/tfenv/bin/* && \
+    chmod 777 ${HOME_DIR}/pkenv/bin/* && \
     rm -r /tmp/* && \
     yum clean all
+ENV HOME=${HOME_DIR}
```
Comment (jaredfholgate) [inline Dockerfile:111]: What is this used for?

File: Dockerfile [hunk 4]
```diff
@@ -97,13 +95,17 @@ RUN yum update -y && \
...
+RUN mkdir ${HOME_DIR}
+COPY .terraformrc ${HOME_DIR}/.terraformrc
 COPY --from=build /go/bin /usr/local/go/bin
-COPY --from=build /src/tfenv /home/runtimeuser/tfenv
-RUN chmod 777 /home/runtimeuser/tfenv && \
-    git clone https://github.com/iamhsa/pkenv.git /pkenv && \
-    cd /pkenv && rm -rf .git && \
+COPY --from=build /src/tfenv ${HOME_DIR}/tfenv
+COPY --from=build /src/pkenv ${HOME_DIR}/pkenv
+RUN mkdir ${HOME_DIR}/tflintenv && \
```
Comment (jaredfholgate) [inline Dockerfile:104]: I guess this is all one line to reduce image layers?

**Review by jaredfholgate — COMMENTED** [review]
LGTM, just some minor comments / questions.

File: avmmakefile (hunk 2 above)
Comment (matt-FFFFFF) [reply avmmakefile:52]: Yeah true. Though the pr-check needs to have a clean base to check that it works ok.

File: Dockerfile (hunk 3 above)
Comment (matt-FFFFFF) [reply Dockerfile:111]: Sets home dir on container to /home/runtimeuser as opposed to your actual home directory from the passwd file, which won't exist

File: Dockerfile (hunk 4 above)
Comment (matt-FFFFFF) [reply Dockerfile:104]: Nope. It's a style error 😊

**Comment by matt-FFFFFF** [comment]
Sorry. Good comments - responded

**Pull request merged**